### PR TITLE
Add proper error handling

### PR DESCRIPTION
**Pull request by AndreyBelym**

### CODE REVIEW

File: src/errors/index.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        super(message(data));
+
+        this.name = constructor.name;
+        this.data = data;
+        this.code = code;
+    }
+}
+
+export class BrowserPathNotSetError extends BasicError {
+    constructor () {
```
Comment (helen-dikareva) [inline src/errors/index.js]: (minor) Can we omit this empty constructor?

Comment (AndreyBelym) [reply src/errors/index.js]: yep

File: src/templates.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+export const BROWSER_PATH_NOT_SET  = () => dedent `
+    Unable to run the browser. The browser path or command template is not specified.
+`;
+
+export const UNABLE_TO_RUN_BROWSERS = ({ path }) => dedent `
+    Unable to run the browser. The file at ${path} does not exist or is not executable.
+`;
+
+export const NATIVE_BINARY_HAS_FAILED = ({ binary, exitCode, output }) => dedent `
+    Process ${binary} failed with exit code ${exitCode}.
```
Comment (dirk-pieterse) [inline src/templates.js]:
```suggestion
    ${binary} process failed with ${exitCode} exit code.
```

File: src/templates.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    ${ output
+        ? PROCESS_OUTPUT({ output })
+        : ''
+    }
+`;
+
+export const UNABLE_TO_ACCESS_AUTOMATION_API = ({ binary }) => dedent `
+    Process ${binary} can't access the Automation API.
```
Comment (dirk-pieterse) [inline src/templates.js]:
```suggestion
    ${binary} process cannot access the Automation API.
```

File: src/templates.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+`;
+
+export const UNABLE_TO_ACCESS_SCREEN_RECORDING_API = ({ binary }) => dedent `
+    Process ${binary} can't access the Screen Recording API.
```
Comment (dirk-pieterse) [inline src/templates.js]:
```suggestion
    ${binary} process cannot access the Screen Recording API.
```

**Pull request merged**